rules/downgrade/src/Rector/FunctionLike/AbstractDowngradeParamTypeDeclarationRector.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Downgrade\Rector\FunctionLike;
+
+use PhpParser\Node;
+use PhpParser\Node\FunctionLike;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\NullableType;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Function_;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Downgrade\Rector\DowngradeRectorTrait;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\TypeDeclaration\Rector\FunctionLike\AbstractTypeDeclarationRector;
+
+abstract class AbstractDowngradeParamTypeDeclarationRector extends AbstractTypeDeclarationRector implements ConfigurableRectorInterface
+{
+    use DowngradeRectorTrait;
+
+    /**
+     * @var string
+     */
+    public const ADD_DOC_BLOCK = '$addDocBlock';
+
+    /**
+     * @var bool
+     */
+    private $addDocBlock = true;
+
+    /**
+     * @param ClassMethod|Function_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->isAtLeastPhpVersion($this->getPhpVersionFeature())) {
+            return $node;
+        }
+
+        if ($node->params === null || $node->params === []) {
+            return null;
+        }
+
+        foreach ($node->params as $position => $param) {
+            $this->refactorParam($param, $node, (int) $position);
+        }
+
+        return null;
+    }
+
+    public function configure(array $configuration): void
+    {
+        $this->addDocBlock = $configuration[self::ADD_DOC_BLOCK] ?? true;
+    }
+
+    /**
+     * Name of the type to remove
+     */
+    abstract protected function getParamTypeName(): string;
+
+    protected function getRectorDefinitionDescription(): string
+    {
+        return sprintf('Remove the \'%s\' param type, add a @param tag instead', $this->getParamTypeName());
+    }
+
+    /**
+     * @param ClassMethod|Function_ $functionLike
+     */
+    private function refactorParam(Param $param, FunctionLike $functionLike, int $position): void
+    {
+        if ($this->shouldSkipParam($param, $functionLike, $position)) {
+            return;
+        }
+
+        if ($this->addDocBlock) {
+            $node = $functionLike;
+            /** @var PhpDocInfo|null $phpDocInfo */
+            $phpDocInfo = $node->getAttribute(AttributeKey::PHP_DOC_INFO);
+            if ($phpDocInfo === null) {
+                $phpDocInfo = $this->phpDocInfoFactory->createEmpty($node);
+            }
+
+            $type = $this->staticTypeMapper->mapPhpParserNodePHPStanType($param->type);
```

Comment:
```bash
------------------------------------------------------------------------
 rules/downgrade/src/Rector/FunctionLike/AbstractDowngradeParamTypeDeclarationRector.php:86
 ------------------------------------------------------------------------
 - '#Parameter \#1 \$node of method Rector\\StaticTypeMapper\\StaticTypeMapper\:\:mapPhpParserNodePHPStanType\(\) expects PhpParser\\Node, PhpParser\\Node\\Identifier\|PhpParser\\Node\\Name\|PhpParser\\Node\\NullableType\|PhpParser\\Node\\UnionType\|null given#'
 ------------------------------------------------------------------------
```

It means, the `$param->type` can be null here, which would lead to crash

Reply:
Fixed, checking param is not `null` before logic